src/coreclr/vm/loongarch64/thunktemplates.S

```diff
@@ -0,0 +1,38 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "unixasmmacros.inc"
+#include "asmconstants.h"
+
+LEAF_ENTRY StubPrecodeCode
+    pcaddi  $r21, 0x1004  //4, for Type encoding.
+    ld.d  $t2,$r21, (StubPrecodeData__MethodDesc - 4*4)
+    ld.d  $r21,$r21,  (StubPrecodeData__Target - 4*4)
+    jirl  $r0,$r21,0
+LEAF_END_MARKED StubPrecodeCode
+
+LEAF_ENTRY FixupPrecodeCode
+    pcaddi  $r21, 0x1003  //3, for Type encoding.
+    ld.d  $r21,$r21, (FixupPrecodeData__Target - 4*3)
```

Comment:
Just to make sure I understand this code - is the page size of the processor 16kB?

Reply:
Yes, you are right.

I had tested this PR based on the lastest runtime yesterday and it is ok which based on your refactor.